ncm-dpmlfc/src/main/perl/DPMLFC/RuleBasedEditor.pm

```diff
@@ -0,0 +1,778 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+#
+#
+# This module implement a rule-based editor that is used to modify the content
+# of an existing file without taking care of the whole file. Each rule
+# driving the edition process is applied to one matching line. The input for
+# updating the file is the Quattor configuration and conditions can be defined
+# based on the contents of this configuration.
+#
+#######################################################################
+
+package NCM::Component::DPMLFC::RuleBasedEditor;
+
+use strict;
+use NCM::Component;
+use vars qw(@ISA $EC);
+@ISA = qw(NCM::Component);
+$EC=LC::Exception::Context->new->will_store_all;
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use File::Path;
+use File::Copy;
+use File::Compare;
+use File::Basename;
+use File::stat;
+
+use LC::Check;
+use CAF::FileWriter;
+use CAF::FileEditor;
+use CAF::FileReader;
+use CAF::Process;
+use CAF::Object;
+
+use Encode qw(encode_utf8);
+use Fcntl qw(SEEK_SET);
+
+local(*DTA);
+
+
+# Constants use to format lines in configuration files
+# Keep in sync with ncm-xrootd
+use constant LINE_FORMAT_PARAM => 1;
```

Comment:
use `Readonly`, to handle exports, see e.g. https://github.com/quattor/configuration-modules-core/blob/master/ncm-systemd/src/main/perl/Systemd/Service.pm#L20 (the trick is to use `our` for exported readonly)